src/dotnet/commands/dotnet-publish/PublishCommand.cs

```diff
@@ -128,7 +128,7 @@ private bool PublishProjectContext(ProjectContext context, string buildBasePath,
             // Use a library exporter to collect publish assets
             var exporter = context.CreateExporter(configuration);
 
-            var isPortable = string.IsNullOrEmpty(context.RuntimeIdentifier);
+            var isPortable = context.PlatformLibrary != null;
```

Comment:
Please, create a IsPortable property in the context, even if it is an extension, so that we don't leak this knowledge of what is portable or not outside of the context.